txtorcon/torcontrolprotocol.py

```diff
@@ -616,7 +619,8 @@ def _safecookie_authchallenge(self, reply):
         """
         Callback on AUTHCHALLENGE SAFECOOKIE
         """
-
+        if(self._cookie_data is None):
+            raise RuntimeError("Cookie data not read.")
```

Comment:
In Python "if"s are usually written ``if self._cookie_data is None:`` (i.e. without the brackets). +1 on _cookie_data :) thanks.